jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3177,12 +3221,31 @@ public Method getEnclosingMethod() throws SecurityException {
  * @see #isLocalClass()
  * @see #isMemberClass()
  */
+
 @CallerSensitive
 public Class<?> getEnclosingClass() throws SecurityException {
 	Class<?> enclosingClass = getDeclaringClass();
 	if (enclosingClass == null) {
-		enclosingClass = getEnclosingObjectClass();
+		if (cachedEnclosingClass == null) {
+			if (cachedEnclosingClassOffset == -1) {
+				long localCachedEnclosingClassOffset;
+				try {
+					Field cachedEnclosingClassField = Class.class.getDeclaredField("cachedEnclosingClass"); //$NON-NLS-1$
+					localCachedEnclosingClassOffset = getUnsafe().objectFieldOffset(cachedEnclosingClassField);
+				} catch (NoSuchFieldException e) {
+					throw newInternalError(e);
+				}
+			/*[IF Sidecar19-SE]
+			getUnsafe().putObjectRelease(this, cachedEnclosingClassOffset, getEnclosingObjectClass());
+			/*[ELSE]*/
+			getUnsafe().putOrderedObject(this, cachedEnclosingClassOffset, getEnclosingObjectClass());
```

Comment:
Similarly it should be `localCachedEnclosingClassOffset` instead of `cachedEnclosingClassOffset`.
Also, pls add indents to align the code in same block.